### PR TITLE
refactoring linguist for increasing code re-usability in projectionist

### PR DESCRIPTION
**Pull request by anuragc617**
Refactoring the linguist classes to reuse them in projectionist. Projectionist PR for re-using linguist gem https://github.com/github/projectionist/pull/2
## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

File: lib/linguist/heuristics.rb [hunk 1]
```diff
@@ -4,7 +4,6 @@ module Linguist
   # A collection of simple heuristics that can be used to better analyze languages.
   class Heuristics
     HEURISTICS_CONSIDER_BYTES = 50 * 1024
-
```
Comment (lildude) [inline lib/linguist/heuristics.rb:7]: I like this empty line 😁. Can we have it back please.

File: lib/linguist/language.rb [hunk 2]
```diff
@@ -27,13 +27,10 @@ class Language
     @name_index         = {}
     @alias_index        = {}
     @language_id_index  = {}
-
```
Comment (lildude) [inline lib/linguist/language.rb:30]: And this one. Please can we have it back.

File: lib/linguist/language.rb [hunk 3]
```diff
@@ -262,10 +259,9 @@ def initialize(attributes = {})
       @name = attributes[:name] || raise(ArgumentError, "missing name")
 
       @fs_name = attributes[:fs_name]
-
```
Comment (lildude) [inline lib/linguist/language.rb:265]: And me.

**Review by lildude — CHANGES_REQUESTED** [review]
Makes sense to me for the added extensibility it provides. Can we have our empty lines back first 🙏

File: lib/linguist/language.rb (hunk 3 above)
Comment (anuragc617) [reply lib/linguist/language.rb:265]: done

File: lib/linguist/language.rb (hunk 2 above)
Comment (anuragc617) [reply lib/linguist/language.rb]: done

File: lib/linguist/heuristics.rb (hunk 1 above)
Comment (anuragc617) [reply lib/linguist/heuristics.rb]: done

**Review by lildude — APPROVED** [review]

**Pull request merged**